google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/util/AppEngineUtil.java

```diff
@@ -16,13 +16,72 @@
 
 package com.google.cloud.tools.intellij.appengine.util;
 
+import com.intellij.facet.Facet;
+import com.intellij.facet.FacetManager;
+import com.intellij.openapi.module.Module;
+import com.intellij.openapi.project.Project;
+import com.intellij.packaging.artifacts.Artifact;
+import com.intellij.packaging.artifacts.ArtifactManager;
+import com.intellij.packaging.impl.artifacts.ArtifactUtil;
+
+import java.util.Collections;
+import java.util.Set;
+
 /**
  * App Engine utility methods.
  */
 public class AppEngineUtil {
 
+  private static final String APP_ENGINE_STANDARD_FACET_NAME = "Google App Engine";
+
   private AppEngineUtil() {
     // Not designed for instantiation
   }
 
+  /**
+   * A project is an App Engine standard project if it has an artifact that has both an App Engine
+   * standard facet and is an App Engine standard artifact type.
+   */
+  public static boolean isAppEngineStandardProject(final Project project) {
+    Artifact[] artifacts = ArtifactManager.getInstance(project).getArtifacts();
+
+    for (Artifact artifact : artifacts) {
+      if (hasAppEngineStandardFacet(project, artifact)
+          && isAppEngineStandardArtifactType(artifact)) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  /**
+   * An artifact has an app engine standard facet if it has a module who's facet name matches that
+   * of the facet configured by the App Engine legacy IJ plugin.
+   */
+  public static boolean hasAppEngineStandardFacet(Project project, Artifact artifact) {
+    Set<Module> modules = ArtifactUtil
+        .getModulesIncludedInArtifacts(Collections.singletonList(artifact), project);
+
+    for (Module module : modules) {
+      for (Facet facet : FacetManager.getInstance(module).getAllFacets()) {
+        if (facet != null && APP_ENGINE_STANDARD_FACET_NAME.equals(facet.getName())) {
+          return true;
+        }
+      }
+    }
+
+    return false;
+  }
+
+  public static boolean isAppEngineStandardArtifactType(Artifact artifact) {
+    String artifactId = artifact.getArtifactType().getId();
+    return "exploded-war".equalsIgnoreCase(artifactId);
```

Comment:
Actually maybe the naming is ok. What I am implying is thats its "the type of artifact needed to be considered an AE _ app", not that if it has that type then it must be an AE app.